src/core/issues-notes.js

```diff
@@ -146,50 +170,89 @@ function handleIssues($ins, ghIssues, conf) {
   }
 }
 
-async function fetchIssuesFromGithub({ githubAPI }) {
-  const issues = [];
-  const issueNumbers = [...document.querySelectorAll(".issue[data-number]")]
-    .map(elem => Number.parseInt(elem.dataset.number, 10))
-    .filter(number => number);
-  for (const issueNumber of issueNumbers) {
-    let issue; // depends on how fetching goes...
+//derives the text-color-class based on the illumination score of the labelColor
+function deriveTextColorClass(hexColor) {
+  const rgb = parseInt(hexColor, 16);
+  if (isNaN(rgb)) {
+    throw new TypeError(`Invalid hex color: ${hexColor}`);
+  }
+  const red = (rgb >> 16) & 0xff;
+  const green = (rgb >> 8) & 0xff;
+  const blue = (rgb >> 0) & 0xff;
+  const illumination = 0.2126 * red + 0.7152 * green + 0.0722 * blue;
+  return illumination > 140 ? "light" : "dark";
+}
+
+function createLabel(label) {
+  const { color, href, name } = label;
+  let textColorClass = "dark";
+  try {
+    textColorClass = deriveTextColorClass(color);
+  } catch (err) {
+    console.error(err);
+  }
+  const cssClasses = `respec-gh-label respec-label-${textColorClass}`;
+  const style = `background-color: #${color}`;
+  return hyperHTML`<a
+    class="${cssClasses}"
+    style="${style}"
+    href="${href}">${name}</a>`;
+}
+
+async function processResponse(response, issueNumber) {
+  const issue = { title: "", number: issueNumber, state: "" };
+  const jsonPromise = response.json();
+  if (!response.ok) {
+    const errJson = { message: "" };
+    try {
+      Object.assign(errJson, await jsonPromise);
+    } catch (err) {
+      errJson.message = `Error JSON parsing issue #${issueNumber} from GitHub.`;
+    } finally {
+      const msg = `Error fetching issue #${issueNumber} from GitHub. ${issue.message} (HTTP Status ${response.status}).`;
+      pub("error", msg);
+    }
+  } else {
     try {
+      Object.assign(issue, await jsonPromise);
+    } catch (err) {
+      const msg = `Error JSON parsing issue #${issueNumber} from GitHub.`;
+      pub("error", msg);
+    }
+  }
+  return [issueNumber, issue];
+}
+
+async function fetchAndStoreGithubIssues(githubAPI) {
```

Comment:
Does the `AndStore` part means the data will be stored in a map? If then we probably skip that part of the name, as it sounds the function stores data somewhere globally.

Reply:
@sidvishnoi is working on a IDBCache over at:
https://github.com/w3c/respec/pull/1555/files#diff-aa6009161df81293880ab4f7c52d9748

Idea is to land that, and then use it while we wait for support of the Cache API to land in Safari. Unfortunately, we won't be able to use service workers, because cross origin limitations. But we can at least throw responses into the cache.